driver-core/src/test/functional/com/mongodb/client/model/expressions/ArrayExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,275 @@
+    // (Incomplete)
+
+    private final ArrayExpression<IntegerExpression> array123 = ofIntegerArray(1, 2, 3);
+    private final ArrayExpression<BooleanExpression> arrayTTF = ofBooleanArray(true, true, false);
+
+    @Test
+    public void literalsTest() {
+        // Boolean
+        assertExpression(
+                Arrays.asList(true, true, false),
+                arrayTTF,
+                "[true, true, false]");
+        // Integer
+        assertExpression(
+                Arrays.asList(1, 2, 3),
+                array123,
+                "[1, 2, 3]");
+        assertExpression(
+                Arrays.asList(1L, 2L, 3L),
+                ofIntegerArray(1L, 2L, 3L),
+                "[{'$numberLong': '1'}, {'$numberLong': '2'}, {'$numberLong': '3'}]");
+        // Number
+        assertExpression(
+                Arrays.asList(1.0, 2.0, 3.0),
+                ofNumberArray(1.0, 2.0, 3.0),
+                "[1.0, 2.0, 3.0]");
+        assertExpression(
+                Arrays.asList(Decimal128.parse("1.0")),
+                ofNumberArray(Decimal128.parse("1.0")),
+                "[{'$numberDecimal': '1.0'}]");
+        // String
+        assertExpression(
+                Arrays.asList("a", "b", "c"),
+                ofStringArray("a", "b", "c"),
+                "['a', 'b', 'c']");
+        // Date
+        assertExpression(
+                Arrays.asList(Instant.parse("2007-12-03T10:15:30.00Z")),
+                ofDateArray(Instant.parse("2007-12-03T10:15:30.00Z")),
+                "[{'$date': '2007-12-03T10:15:30.00Z'}]");
+        // Document
+        // ...
+
+        // Array
+        ArrayExpression<ArrayExpression<Expression>> arrays = ofArray(ofArray(), ofArray());
+        assertExpression(
+                Arrays.asList(Collections.emptyList(), Collections.emptyList()), arrays,
+                "[[], []]");
+
+        // Mixed
+        ArrayExpression<Expression> expression = ofArray(of(1), of(true), ofArray(of(1.0), of(1)));
+        assertExpression(
+                Arrays.asList(1, true, Arrays.asList(1.0, 1)),
+                expression,
+                "[1, true, [1.0, 1]]");
+    }
+
+    @Test
+    public void filterTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/filter/
+        assertExpression(
+                Stream.of(true, true, false)
+                        .filter(v -> v).collect(Collectors.toList()),
+                arrayTTF.filter(v -> v),
+                // MQL:
+                "{'$filter': {'input': [true, true, false], 'cond': '$$this'}}");
+    }
+
+    @Test
+    public void mapTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/map/
+        assertExpression(
+                Stream.of(true, true, false)
+                        .map(v -> !v).collect(Collectors.toList()),
+                arrayTTF.map(v -> v.not()),
+                // MQL:
```

Comment:
We may want to re-review the comments left in the code (currently, they are mostly in the test code) to see if they are helpful.

Some examples of questionable comments:
- `(Incomplete)` at the top of `ArrayExpressionsFunctionalTest`.
- `// MQL:` when passing a value of the `expectedMql` parameter.